src/io.cpp

```diff
@@ -618,45 +618,105 @@ GeneratedPipelineInput::GeneratedPipelineInput(const Catalog &catalog,
     }
 }
 
+
+Attitude CoverageTest(std::default_random_engine* pReng) { //CoverageTest() takes a Pointer as the paramter now
```

Comment:
Should declare in `io.hpp` if this function is part of the public interface. Otherwise, make it `static` (i.e., private to the file)